File: Neetcode/RightViewBT.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if root is None:
            return []
        ans = []
        dq = deque()
        dq.append(root)
        while dq:
            ans.append(dq[-1].val)
            for _ in range(len(dq)):
                node = dq.popleft()
                if node.left is not None:
                    dq.append(node.left)
                if node.right is not None:
                    dq.append(node.right)
        return ans

File: Neetcode/RightViewBT.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxH = -1;
    vector<int> view;
    void bt(TreeNode* node, int h){
        if (node == nullptr){
            return;
        }
        if (h > maxH){
            view.push_back(node->val);
            maxH = h;
        }
        bt(node->right, h+1);
        bt(node->left, h+1);
    }
    vector<int> rightSideView(TreeNode* root) {
        bt(root, 0);
        return view;
    }
};
